python/benchmark/test_gen_data.py

```diff
@@ -103,3 +103,91 @@ def test_make_low_rank_matrix(dtype: str) -> None:
 
         _, s, _ = svd(X)
         assert sum(s) - 5 < 0.1, "X rank is not approximately 5"
+
+
+@pytest.mark.parametrize("dtype", ["float32", "float64"])
+def test_make_regression_low_rank(dtype: str) -> None:
+    # Effective rank = 5
```

Comment:
Seems you can combine test_make_regression_low_rank and test_make_regression_well_conditioned with effictive_rank parameterized.

Reply:
To invoke the well-conditioned test, I need to leave off "--effective_rank" entirely; the parser expects either an int or no arg at all. I could combine them into one test, but that would still require defining two sets of input_args and having an if statement.

Reply:
You can conditionally construct the args list from multiple pieces depending on the value of effective_rank parameter in a parameterized test.